modules/dnn/src/caffe/caffe.proto

```diff
@@ -62,6 +62,12 @@ message BlobProto {
   repeated double double_data = 8 [packed = true];
   repeated double double_diff = 9 [packed = true];
 
+  enum DataType {
```

Comment:
it would be nice to put something like that into official Caffe tree, or maybe find a branch where it's already done

Reply:
NVidia's Caffe supports models stored this way, https://github.com/NVIDIA/caffe/blob/caffe-0.16/src/caffe/proto/caffe.proto#L9. So `shrinkCaffeModel` designed now to store float16 weights in field with the same tag.